File: Map/Z_InterpretorWCDL/src/expression/VarExpression.java
package expression;

import domain.DomainException;
import domain.IDictionary;
import domain.IHeap;

public class VarExpression implements IExpression
{
	private String varName;
	
	public VarExpression(String variableName)
	{
		varName = variableName;
	}
	
	@Override
	public int eval(IDictionary<String, Integer> symbolTable, IHeap<Integer,Integer> heap) 
	{
		if(symbolTable.contains(varName))
			return symbolTable.getValue(varName);
		throw new DomainException("Variable not found!");
	}
	
	@Override 
	public String toString()
	{
		return varName;
	}

}

File: Map/Map/Map/expression/VarExpression.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Map.domain;

namespace Map.expression
{
    class VarExpression : IExpression
    {
        private String varName;

        public VarExpression(String name)
        {
            varName = name;
        }

        public int eval(IDict<string, int> symbolTable)
        {
            if (symbolTable.contains(varName))
                return symbolTable.getValue(varName);
            throw new Exception("Variable not found!");
        }

        public override string ToString()
        {
            return varName;
        }
    }
}
